syntax = "proto3";

message Voting {

  enum VoteType{
    ACTIVE = 0;
    NOT_ACTIVE = 1;
  }
  enum VoteResult{
    WIN = 0;
    LOSE = 1;
    UNKNOWN = 2;
  }
  message Vote {
    string issuer_public_key = 1;
    bool accepted = 2;
    uint64 timestamp = 3;
    string transaction_id = 4;
  }

  string elector_public_key = 1;
  string data = 2;
  VoteResult vote_result = 3;
  uint64 close_vote_timestamp = 4;
  VoteType voteType = 5;
  repeated  Vote vote = 6;
  uint64 timestamp = 7;
  string transaction_id = 8;
}


message VotingContainer {
  repeated Voting entries = 1;
}